peer/set.go

```diff
@@ -0,0 +1,71 @@
+package peer
+
+import (
+	"sync"
+)
+
+// PeerSet is a threadsafe set of peers
+type Set struct {
+	lk sync.RWMutex
+	ps map[ID]struct{}
+
+	size int
```

Comment:
the size field seems unused.

Reply:
The field is used internally in `TryAdd()`.